mlflow/utils/docstring_utils.py

```diff
@@ -176,3 +185,46 @@ def decorator(func):
 """,
     }
 )
+
+
+def docstring_version_compatibility_warning(integration_name):
+    """
+    Generates a docstring that can be applied as a note stating a version compatibility range for
+    a given flavor.
+
+    :param integration_name: The name of the module as stored within ml-package-versions.yml
+    :return: The wrapped function with the additional docstring header applied
+    """
+
+    def get_version_ranges(module_key):
+        versions = _ML_PACKAGE_VERSIONS[module_key]["models"]
+        min_version = versions["minimum"]
+        max_version = versions["maximum"]
+        return min_version, max_version
```

Comment:
Can we move this function in the top level?

Reply:
Excellent idea. renamed and added usage notes